apps/passwd.c

```diff
@@ -343,30 +331,48 @@ static char *md5crypt(const char *passwd, const char *magic, const char *salt)
 
     out_buf[0] = 0;
     magic_len = strlen(magic);
+    OPENSSL_strlcpy(ascii_magic, magic, sizeof ascii_magic);
+#ifdef CHARSET_EBCDIC
+    if ((magic[0] & 0x80) != 0)    /* High bit is 1 in EBCDIC alnums */
```

Comment:
Wouldn't it be safer to do the EBCDIC to ASCII conversion unconditionally here?  Or is this an attempt to automatically detect the input character set?

Just testing the first character's high bit misses some printables and captures some non alnum ones too.